addons/website_project/controllers/main.py

```diff
@@ -35,4 +35,14 @@ def extract_data(self, model, values):
             data['record']['email_from'] = values['email_from']
             if not partner_id:
                 data['record']['email_cc'] = values['email_from']
+                data['record']['partner_name'] = values['partner_phone']
+                data['record']['partner_name'] = values['partner_name']
+                data['record']['partner_company_name'] = values['partner_company_name']
+            else:
+                if not partners_list[0].phone:
+                    data['record']['partner_name'] = values['partner_phone']
+                if not partners_list[0].name:
+                    data['record']['partner_name'] = values['partner_name']
+                if not partners_list[0].company_name:
+                    data['record']['partner_company_name'] = values['partner_company_name']
```

Comment:
We could change a bit the code to get the `partner` recordset instead of each time doing `partners_list[0]...`